eZ/Publish/Core/FieldType/Image/IO/OptionsProvider.php

```diff
@@ -8,61 +8,42 @@
  */
 namespace eZ\Publish\Core\FieldType\Image\IO;
 
+use eZ\Publish\Core\MVC\ConfigResolverInterface;
+
 class OptionsProvider
 {
-    protected $varDir;
-
-    protected $storageDir;
-
-    protected $draftImagesDir;
-
-    protected $publishedImagesDir;
+    /** @var \eZ\Publish\Core\MVC\ConfigResolverInterface */
+    protected $configResolver;
 
-    public function __construct(array $values = [])
+    public function __construct(ConfigResolverInterface $configResolver)
     {
-        $this->varDir = isset($values['var_dir']) ? $values['var_dir'] : null;
-        $this->storageDir = isset($values['storage_dir']) ? $values['storage_dir'] : null;
-        $this->draftImagesDir = isset($values['draft_images_dir']) ? $values['draft_images_dir'] : null;
-        $this->publishedImagesDir = isset($values['published_images_dir']) ? $values['published_images_dir'] : null;
+        $this->configResolver = $configResolver;
     }
 
-    public function setVarDir($varDir)
+    protected function getSetting(string $name): ?string
```

Comment:
Also here, is it should be protected function? There are no extends and in "Core" there is no BC guarantee afaik